x/datadeal/keeper/deal.go

```diff
@@ -230,3 +241,94 @@ func (k Keeper) GetAllDataSaleList(ctx sdk.Context) ([]types.DataSale, error) {
 
 	return dataSales, nil
 }
+
+func (k Keeper) VoteDataVerification(ctx sdk.Context, vote *types.DataVerificationVote, signature []byte) error {
+	if err := vote.ValidateBasic(); err != nil {
+		return sdkerrors.Wrapf(types.ErrDataVerificationVote, err.Error())
+	}
+
+	sellerAcc, err := sdk.AccAddressFromBech32(vote.SellerAddress)
```

Comment:
I think that oracle will sign with the voter key rather than the seller's private key. Please check it and if it is correct, change it to the voter address.

Reply:
Ah, yes. I missed it.